pkg/controllers/rollout/controller.go

```diff
@@ -503,6 +554,11 @@ func determineBindingsToUpdate(
 	if boundingCandidatesUnselectedIndex < len(boundingCandidates) {
 		staleUnselectedBinding = append(staleUnselectedBinding, boundingCandidates[boundingCandidatesUnselectedIndex:]...)
 	}
+	// Append the pending removal bindings to the staleUnselectedBinding list.
+	//
+	// This is necessary as these bindings might also need a status update (e.g., their apply strategies
+	// have been updated).
+	staleUnselectedBinding = append(staleUnselectedBinding, pendingRemovalBindingList...)
```

Comment:
the updateStaleBindingsStatus function currently only update scheduled and bound one (line 863). But the remove candidates are "BindingStateUnscheduled"

Reply:
Hi Ryan! That method has been updated to process `Unscheduled` bindings as well.